fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/PhysicalDataModel.java

```diff
@@ -339,24 +339,26 @@ public void visit(DataModelVisitor v, final String tagGroup, final String tag, S
         // visit just the matching subset of objects. If a transactionSupplier has been provided, we break up the
         // operation into multiple transactions to avoid transaction size limitations (e.g. with Citus FK creation)
         if (transactionSupplier != null) {
+
+            
             ITransaction tx = transactionSupplier.get();
             try {
                 int count = 0;
                 for (IDatabaseObject obj: allObjects) {
                     if (tag == null || obj.getTags().get(tagGroup) != null && tag.equals(obj.getTags().get(tagGroup))) {
-                        if (++count == 10) {
-                            // commit the current transaction and start a fresh one
-                            tx.close();
-                            tx = transactionSupplier.get();
-                            count = 0;
-                        }
-
                         try {
+                            logger.info("Visiting[" + count + "] " + obj.getName());
                             obj.visit(v);
                         } catch (RuntimeException x) {
                             tx.setRollbackOnly();
                             throw x;
                         }
+                        if (++count == 1) {
```

Comment:
won't this always be true (given that count is initialized to 0 above and then also within this block)?  it used to be 10...

Reply:
Yes, but here I was battling with Citus failing with out of SHM errors and so I was trying to reduce the size of the transaction. This may be just an issue when running with Citus in a docker container because I didn't see this issue when running with Azure Citus. I would like to leave this pattern here for now, because if we are able to address the underlying failure (through Citus configuration, perhaps), then this can be increased again. I'll add a comment to reflect this.